### DISCUSSION TITLE
RoutablePage

### DISCUSSION THREAD

**Discussion by kiyoshi-satoo** [wagtail/wagtail#11475] [Q&A] [answered]
Hi everyone.
I want to make a routablepage but to take file name or title as argument.Since the file name uses spaces(%20) and various other characters I get 404 error.I tried using Regex but to no avail.Maybe I am doing something wrong,so I will be glad if someone can suggest some solution.

**Routable page url**
```
# @re_path(r'^(?P<title>[-\w\s]+)/$') 
@path('<str:title>/')
def screen_by_title(self, request,title):
    print(title)
    return HttpResponse(status=200)
```

**Template**
```
<a href="{% routablepageurl page 'screen_by_title' title=image.alt %}"></a>

After render

<a href="projects/some-project/Screen%201/"></a>
```

**Comment by gasman** [comment, marked as answer]
Unfortunately this is a known issue with RoutablePageMixin (see #3653) - URLs still have to follow the same restrictions as page slugs.

Maybe you could use [the `quote` / `unquote` functions in `django.contrib.admin.utils`](https://github.com/django/django/blob/4879907223d70ee1a82474d9286ccfa5dae96971/django/contrib/admin/utils.py#L87-L99) (and the `admin_urlquote` template filter from the `admin_urls` tag library within templates - `title=image.alt|admin_urlquote`) to convert these strings to slug-safe versions within the URL?

**Reply by kiyoshi-satoo** [reply]
Hey @gasman.Thank you for your reply. I will try to test this option.

For a better understanding - I am trying to do something like [this](https://archive.area17.com/2023_openai/03_research-index%402x.png)
I can create a separate django view for all this, but I would like to keep the page hierarchy using **RoutablePageMixin**.

If you have any other suggestions that would be cool, thanks.
